### PR TITLE
ENH: ExtensionArray.repeat

### PR DESCRIPTION
**Pull request by TomAugspurger**
cc @jreback

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -987,7 +987,8 @@ update the ``ExtensionDtype._metadata`` tuple to match the signature of your
 
 **Other changes**
 
-- ``ExtensionArray`` has gained the abstract methods ``.dropna()`` (:issue:`21185`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.24.0.rst]: This method isn't abstract.

File: pandas/core/arrays/base.py [hunk 2]
```diff
@@ -580,6 +580,33 @@ def factorize(self, na_sentinel=-1):
         uniques = self._from_factorized(uniques, self)
         return labels, uniques
 
+    def repeat(self, repeats, axis=None):
+        """
+        Repeat elements of an array.
+
```
Comment (TomAugspurger) [inline pandas/core/arrays/base.py]: Need a versionadded. Will do that on merge maybe, unless there are other comments on the implementation.

**Comment by TomAugspurger** [comment]
Added a fixture to parametrize the test across ExtensionArray.repeat and
Series.repeat.

I don't think there's any code duplication that could be removed. Series
needs to handle repeating the index, and just dispatches to `._values` for
repeating the array.

**Review by jreback — APPROVED** [review]
these methods are in api.rst?

**Comment by TomAugspurger** [comment]
ExtensionArray is an auto summary, so these will be added automatically.

Though now that I think about it, it won't have the "private" methods like
_from_sequnce,
since numpydoc / autodoc excludes items starting with an `_`. But I guess
they'll be looking
at the source / GitHub anyway.

**Pull request merged**